src/main/java/main/Parser.java

```diff
@@ -0,0 +1,98 @@
+package main;
+
+import Objects.Deadline;
+import Objects.Event;
+import Objects.Task;
+import Objects.Todo;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Scanner;
+
+public class Parser {
+    Storage storage;
+
+    public static void input(TaskList taskList){
+        Scanner sc = new Scanner(System.in);
+        boolean isChanged = false;
+        while (true) {
+            if(isChanged){
+                try {
+                    Storage.writeToFile(taskList);
+                    isChanged = false;
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            String str = sc.nextLine();
+            String[] splitStr = str.split("\\s+",2);
+
+            switch (splitStr[0].toLowerCase()) {
+                case "bye":
+                    Ui.bye();
+                    return;
+                case "list":
+                    taskList.printList();
+                    break;
+                case "mark":
+                    taskList.mark(Integer.parseInt(splitStr[1])-1);
```

Comment:
Should we handle the case where the user enters something other than numbers? `Integer.parseInt()` will throw a `NumberFormatException`.
```suggestion
                    taskList.mark(Integer.parseInt(splitStr[1]) - 1);
```